src/client/dfs/dfs.c

```diff
@@ -846,16 +858,48 @@ open_symlink(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 	return ENOTSUP;
 }
 
+static inline void
+set_daos_iod(bool create, daos_iod_t *iod, char *buf, size_t size)
+{
+	d_iov_set(&iod->iod_name, buf, strlen(buf));
+	iod->iod_nr		= 1;
+	iod->iod_size	= DAOS_REC_ANY;
+	iod->iod_recxs	= NULL;
+	iod->iod_type	= DAOS_IOD_SINGLE;
+
+	if (create) {
+		iod->iod_size = size;
+	}
+}
+
+static void
+create_sb(bool create, daos_iod_t *iods, daos_key_t *dkey)
+{
+	int i = 0;
+
+	d_iov_set(dkey, SB_DKEY, strlen(SB_DKEY));
+
+	set_daos_iod(create, &iods[i++], MAGIC_NAME, sizeof(dfs_magic_t));
+	set_daos_iod(create, &iods[i++], SB_VERSION_NAME, sizeof(dfs_sb_ver_t));
+	set_daos_iod(create, &iods[i++], LAYOUT_NAME, sizeof(dfs_layout_ver_t));
+	set_daos_iod(create, &iods[i++], CS_NAME, sizeof(daos_size_t));
+	set_daos_iod(create, &iods[i++], OC_NAME, sizeof(daos_oclass_id_t));
+	set_daos_iod(create, &iods[i++], FEAT_COMPAT_NAME, sizeof(daos_size_t));
+	set_daos_iod(create, &iods[i++], FEAT_INCO_NAME, sizeof(daos_size_t));
+	set_daos_iod(create, &iods[i++], MKFS_TIME_NAME, sizeof(daos_size_t));
+	set_daos_iod(create, &iods[i++], STATE_NAME, sizeof(daos_size_t));
```

Comment:
As of today, “vos_dfs_sample.yaml” considers that those structures exits. So, in an effort to replicate the current yaml file I tried to add the missing structures. I am guessing that I should not include them since its implementation is not fully supported, and let “vos_dfs_sample.yaml” reflects the current implementation.

The actual size of this PR is not terrible big. Unfortunately, I messed up this branch and now there are other files from others commits that don't belong to my original PR.
So, I created 2947. I guess I am going to close this PR. Sorry for the inconvenience